networkx/algorithms/similarity.py

```diff
@@ -641,15 +643,24 @@ def match_edges(u, v, pending_g, pending_h, Ce, matched_uv=[]):
             #assert C.shape == (m + n, m + n)
 
             # Forbid structurally invalid matches
-            inf = min(min(Ce.C.sum(axis=0)), min(Ce.C.sum(axis=1))) + 1
+            # NOTE: inf remembered from Ce construction
             for k, i in zip(range(m), g_ind):
-                g = pending_g[i]
+                g = pending_g[i][:2]
                 for l, j in zip(range(n), h_ind):
-                    h = pending_h[j]
-                    if not any(g in ((p, u), (u, p)) and h in ((q, v), (v, q))
-                               or g == (u, u) and h == (v, v)
+                    h = pending_h[j][:2]
+                    if nx.is_directed(G2) or nx.is_directed(G2):
```

Comment:
Can you look at this if statement again -- are both supposed to be G2?

Reply:
Uh-oh -- thanks for spotting this!
Please expect the fix commit later today.